domain/model/service/service.go

```diff
@@ -542,3 +589,59 @@ func (s *Service) UpdateCredential(
 
 	return s.st.UpdateCredential(ctx, uuid, key)
 }
+
+// getWatchActivatedModelsMapper returns a mapper function that filters change events to
+// include only those associated with activated models.
+func getWatchActivatedModelsMapper(st State) func(ctx context.Context, db database.TxnRunner,
```

Comment:
At the moment this only exists to form a closure around state which is carried on the service. I would suggest just making this function signature the one being returned and avoiding the closure and everything that goes along with it.

Reply:
Hi tom, the reason for this is because I'm testing the mapper logic in service_test before returning a NewNamespaceMapperWatcher for the mockWatcherFactory. Please do let me know if you have any suggestions for a better approach :)

Reply:
Hi Tom, the reason for this is because I'm testing the mapper logic in service_test before returning a NewNamespaceMapperWatcher for the mockWatcherFactory. Perhaps should I shift the mapper test to another test func?

Reply:
In my opinion I wouldn't be testing this directly. Instead I would be testing the function that uses it to make sure its contract is correct with respect to this. The things that we care about with respect to what this function does are inherited by the caller and format part of the callers contract.

This is an implementation detail that can't change without breaking tests.